java/memory/src/main/java/org/apache/arrow/memory/BaseAllocator.java

```diff
@@ -111,15 +112,15 @@ protected BaseAllocator(
       historicalLog = null;
       childLedgers = null;
     }
-    this.roundingPolicy = roundingPolicy;
+    this.roundingPolicy = config.getRoundingPolicy();
   }
 
   AllocationListener getListener() {
     return listener;
   }
 
   @Override
-  public BufferAllocator getParentAllocator() {
```

Comment:
Just to double-check - is it intentional that we're exposing the concrete implementation now here?

Reply:
IMO, it is useful to expose the hierarchical structure of the allocators.